icicle-cpu/src/lifter/mod.rs

```diff
@@ -702,12 +723,13 @@ impl BlockLifter {
             }
         }
 
+        // TODO: When does this error happen exactly?
```

Comment:
This occurs when the block starts with an invalid instruction there is no need to generate a block at all, and we can immediately raise an exception.